gobblin-core/src/main/java/gobblin/writer/AvroHdfsDataWriter.java

```diff
@@ -104,21 +109,40 @@ public AvroHdfsDataWriter(State properties, String relFilePath, String fileName,
       this.fs.mkdirs(this.outputFile.getParent());
     }
 
+    this.datumWriter = new GenericDatumWriter<GenericRecord>();
     this.writer = createDatumWriter(schema, this.stagingFile, bufferSize, CodecType.valueOf(codecType), deflateLevel);
   }
 
+  /**
+   * This constructor is for subclass AvroHdfsHourlyDataWriter
+   */
+  protected AvroHdfsDataWriter(State properties, int branch) throws IOException {
+    String uri =
+        properties.getProp(
+            ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, branch),
+            ConfigurationKeys.LOCAL_FS_URI);
+    Configuration conf = new Configuration();
+    this.fs = FileSystem.get(URI.create(uri), conf);
+    this.stagingFile = null;
```

Comment:
Given this and the following lines, we need to rethink if inheritance is the best approach here. We need to sync up on the writer part.